criu/mount.c

```diff
@@ -3193,6 +3218,7 @@ static int populate_mnt_ns(void)
 
 	ret = mnt_tree_for_each(pms, do_mount_one);
 	mnt_tree_for_each(pms, do_close_one);
+	mnt_tree_for_each(root_yard_mp, set_unbindable);
```

Comment:
bzzzzz. I don't like this. First, we don't need to do this if ret isn't zero. Second, can we do this without enumerating all mounts?

We can try to resurrect this https://lkml.org/lkml/2017/1/23/712

Reply:
> bzzzzz. I don't like this. First, we don't need to do this if ret isn't zero. Second, can we do this without enumerating all mounts?

Done.

>We can try to resurrect this https://lkml.org/lkml/2017/1/23/712

I don't think it has any connection with unbindable flag, we can set these flag at any moment to any visible mount.

Offtop: I think, generally, that the kernel interface to add an existing mount to an existing shared group (or slave group) is a good thing. But I'm still not sure how to convince MS kernel guys to allow us this.

Reply:
Ret check, special delayed_unbindable list to improve enumeration done.